### ISSUE TITLE
[SOLR/Feature request] Basic authentication

### ISSUE THREAD

**Issue by Kaelnor** [HariSekhon/Nagios-Plugins#240]
Hi,

I was trying to use your plugins for solr core (check_solr_*) but I have an instance which is protected behind basic auth.

I took a look at the solr lib in your repo but it doesn't seem like this is a supported use case.

Am I missing something ? If not, could you consider adding this feature in a future release ?

Thanks a lot for your awesome work !

**Comment by HariSekhon** [comment]
I've just pushed some updates that should enable you to do this.

Please checkout the `solr_basic_auth` branch I've just pushed and try those.

Thanks

Hari

**Comment by Kaelnor** [comment]
Thank you for this quick update.

I tested check_solr_api_ping.pl and check_solr_core.pl with authentication and they are both working. The other solr plugins should work too as it's the same code.

I noticed that one of the scripts is python instead of perl (check_solr_version.py). I took a look at the code and the authentication seems to be an easy fix considering this function exists:

https://github.com/HariSekhon/pylib/blob/049f558580b50c0e93d20801b20d9f555514dcd5/harisekhon/cli.py#L444

I'll try to test a solution tomorrow if I have the time to do so.
